### ISSUE TITLE
Cloud controller manger crashes during startup on Azure Stack because of no zones

### ISSUE THREAD

**Issue by lobziik** [kubernetes-sigs/cloud-provider-azure#715]
**What happened**:
Recent version of Azure cloud controller manager panicing during startup because of no zones support on Azure Stack
Error comes from this line: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/13f66db1f3eb7dd00f7e35d0b281d4ca17965c61/pkg/provider/azure.go#L579

**What you expected to happen**:
It should starts up successfully.

**How to reproduce it**:
Try to run CCM built after https://github.com/kubernetes-sigs/cloud-provider-azure/commit/7a105d9fd572e21dbc8b16af8182a8993d7b2007 this patch on Azure Stack

**Anything else we need to know?**:
https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-network-differences?view=azs-2102 - this doc say that Avalability Zones not yet supported on StackCloud.

**Environment**:
- kube 1.21.1, openshift 4.9
- ccm built from https://github.com/kubernetes-sigs/cloud-provider-azure/commit/61c726d73edb4204d608612fb8c198f493f0995d this revision

**Comment by feiskyer** [comment]
@lobziik could you share the error logs? I think we could skip the sync logic when the API detects zone is not supported.

**Comment by lobziik** [comment]
@feiskyer, yup, think the only way is to skip this check for Stack.

Log, stripped a bit: 
```
I0721 13:04:43.576780  441897 flags.go:59] FLAG: --add_dir_header="false"
I0721 13:04:43.576823  441897 flags.go:59] FLAG: --address="0.0.0.0"
I0721 13:04:43.576828  441897 flags.go:59] FLAG: --allocate-node-cidrs="false"
I0721 13:04:43.576834  441897 flags.go:59] FLAG: --allow-untagged-cloud="false"
I0721 13:04:43.576837  441897 flags.go:59] FLAG: --alsologtostderr="false"
I0721 13:04:43.576840  441897 flags.go:59] FLAG: --authentication-kubeconfig=""
I0721 13:04:43.576843  441897 flags.go:59] FLAG: --authentication-skip-lookup="false"
I0721 13:04:43.576847  441897 flags.go:59] FLAG: --authentication-token-webhook-cache-ttl="10s"
I0721 13:04:43.576854  441897 flags.go:59] FLAG: --authentication-tolerate-lookup-failure="false"
I0721 13:04:43.576858  441897 flags.go:59] FLAG: --authorization-always-allow-paths="[/healthz,/readyz,/livez]"
I0721 13:04:43.576868  441897 flags.go:59] FLAG: --authorization-kubeconfig=""
I0721 13:04:43.576873  441897 flags.go:59] FLAG: --authorization-webhook-cache-authorized-ttl="10s"
I0721 13:04:43.576879  441897 flags.go:59] FLAG: --authorization-webhook-cache-unauthorized-ttl="10s"
I0721 13:04:43.576884  441897 flags.go:59] FLAG: --bind-address="127.0.0.1"
I0721 13:04:43.576889  441897 flags.go:59] FLAG: --cert-dir=""
I0721 13:04:43.576895  441897 flags.go:59] FLAG: --cidr-allocator-type="RangeAllocator"
I0721 13:04:43.576901  441897 flags.go:59] FLAG: --client-ca-file=""
I0721 13:04:43.576907  441897 flags.go:59] FLAG: --cloud-config="/tmp/tel-qrzafe31/fs/etc/kubernetes-cloud-config/cloud.conf"
I0721 13:04:43.576915  441897 flags.go:59] FLAG: --cloud-config-key="cloud-config"
I0721 13:04:43.576921  441897 flags.go:59] FLAG: --cloud-config-secret-name="azure-cloud-provider"
I0721 13:04:43.576927  441897 flags.go:59] FLAG: --cloud-config-secret-namespace="kube-system"
I0721 13:04:43.576933  441897 flags.go:59] FLAG: --cloud-provider="azure"
I0721 13:04:43.576939  441897 flags.go:59] FLAG: --cluster-cidr=""
I0721 13:04:43.576944  441897 flags.go:59] FLAG: --cluster-name="dmoiseev-bzs5m"
I0721 13:04:43.576950  441897 flags.go:59] FLAG: --concurrent-service-syncs="1"
I0721 13:04:43.576957  441897 flags.go:59] FLAG: --configure-cloud-routes="false"
I0721 13:04:43.576962  441897 flags.go:59] FLAG: --contention-profiling="false"
I0721 13:04:43.576968  441897 flags.go:59] FLAG: --controller-start-interval="0s"
I0721 13:04:43.576976  441897 flags.go:59] FLAG: --controllers="[*,-cloud-node,-route]"
I0721 13:04:43.576985  441897 flags.go:59] FLAG: --enable-dynamic-reloading="false"
I0721 13:04:43.576991  441897 flags.go:59] FLAG: --enable-leader-migration="false"
I0721 13:04:43.576996  441897 flags.go:59] FLAG: --external-cloud-volume-plugin=""
I0721 13:04:43.577002  441897 flags.go:59] FLAG: --feature-gates=""
I0721 13:04:43.577010  441897 flags.go:59] FLAG: --help="false"
I0721 13:04:43.577015  441897 flags.go:59] FLAG: --http2-max-streams-per-connection="0"
I0721 13:04:43.577022  441897 flags.go:59] FLAG: --kube-api-burst="30"
I0721 13:04:43.577028  441897 flags.go:59] FLAG: --kube-api-content-type="application/vnd.kubernetes.protobuf"
I0721 13:04:43.577035  441897 flags.go:59] FLAG: --kube-api-qps="20"
I0721 13:04:43.577042  441897 flags.go:59] FLAG: --kubeconfig="kubeconfig"
I0721 13:04:43.577050  441897 flags.go:59] FLAG: --leader-elect="true"
I0721 13:04:43.577057  441897 flags.go:59] FLAG: --leader-elect-lease-duration="15s"
I0721 13:04:43.577063  441897 flags.go:59] FLAG: --leader-elect-renew-deadline="10s"
I0721 13:04:43.577069  441897 flags.go:59] FLAG: --leader-elect-resource-lock="leases"
I0721 13:04:43.577074  441897 flags.go:59] FLAG: --leader-elect-resource-name="cloud-controller-manager"
I0721 13:04:43.577081  441897 flags.go:59] FLAG: --leader-elect-resource-namespace="openshift-cloud-controller-manager"
I0721 13:04:43.577087  441897 flags.go:59] FLAG: --leader-elect-retry-period="2s"
I0721 13:04:43.577092  441897 flags.go:59] FLAG: --leader-migration-config=""
I0721 13:04:43.577101  441897 flags.go:59] FLAG: --log-flush-frequency="5s"
I0721 13:04:43.577107  441897 flags.go:59] FLAG: --log_backtrace_at=":0"
I0721 13:04:43.577115  441897 flags.go:59] FLAG: --log_dir=""
I0721 13:04:43.577128  441897 flags.go:59] FLAG: --log_file=""
I0721 13:04:43.577134  441897 flags.go:59] FLAG: --log_file_max_size="1800"
I0721 13:04:43.577140  441897 flags.go:59] FLAG: --logtostderr="true"
I0721 13:04:43.577146  441897 flags.go:59] FLAG: --master=""
I0721 13:04:43.577150  441897 flags.go:59] FLAG: --min-resync-period="12h0m0s"
I0721 13:04:43.577156  441897 flags.go:59] FLAG: --node-cidr-mask-size="24"
I0721 13:04:43.577162  441897 flags.go:59] FLAG: --node-cidr-mask-size-ipv4="0"
I0721 13:04:43.577168  441897 flags.go:59] FLAG: --node-cidr-mask-size-ipv6="0"
I0721 13:04:43.577174  441897 flags.go:59] FLAG: --node-monitor-period="5s"
I0721 13:04:43.577179  441897 flags.go:59] FLAG: --node-status-update-frequency="5m0s"
I0721 13:04:43.577184  441897 flags.go:59] FLAG: --node-sync-period="0s"
I0721 13:04:43.577190  441897 flags.go:59] FLAG: --one_output="false"
I0721 13:04:43.577195  441897 flags.go:59] FLAG: --permit-address-sharing="false"
I0721 13:04:43.577201  441897 flags.go:59] FLAG: --permit-port-sharing="false"
I0721 13:04:43.577207  441897 flags.go:59] FLAG: --port="0"
I0721 13:04:43.577213  441897 flags.go:59] FLAG: --profiling="true"
I0721 13:04:43.577221  441897 flags.go:59] FLAG: --requestheader-allowed-names="[]"
I0721 13:04:43.577234  441897 flags.go:59] FLAG: --requestheader-client-ca-file=""
I0721 13:04:43.577240  441897 flags.go:59] FLAG: --requestheader-extra-headers-prefix="[x-remote-extra-]"
I0721 13:04:43.577249  441897 flags.go:59] FLAG: --requestheader-group-headers="[x-remote-group]"
I0721 13:04:43.577257  441897 flags.go:59] FLAG: --requestheader-username-headers="[x-remote-user]"
I0721 13:04:43.577264  441897 flags.go:59] FLAG: --route-reconciliation-period="10s"
I0721 13:04:43.577270  441897 flags.go:59] FLAG: --secure-port="10258"
I0721 13:04:43.577276  441897 flags.go:59] FLAG: --service-cluster-ip-range=""
I0721 13:04:43.577282  441897 flags.go:59] FLAG: --skip_headers="false"
I0721 13:04:43.577288  441897 flags.go:59] FLAG: --skip_log_headers="false"
I0721 13:04:43.577294  441897 flags.go:59] FLAG: --stderrthreshold="2"
I0721 13:04:43.577300  441897 flags.go:59] FLAG: --tls-cert-file=""
I0721 13:04:43.577306  441897 flags.go:59] FLAG: --tls-cipher-suites="[]"
I0721 13:04:43.577313  441897 flags.go:59] FLAG: --tls-min-version=""
I0721 13:04:43.577319  441897 flags.go:59] FLAG: --tls-private-key-file=""
I0721 13:04:43.577324  441897 flags.go:59] FLAG: --tls-sni-cert-key="[]"
I0721 13:04:43.577331  441897 flags.go:59] FLAG: --use-service-account-credentials="true"
I0721 13:04:43.577337  441897 flags.go:59] FLAG: --v="4"
I0721 13:04:43.577343  441897 flags.go:59] FLAG: --version="false"
I0721 13:04:43.577353  441897 flags.go:59] FLAG: --vmodule=""
I0721 13:04:44.094068  441897 serving.go:347] Generated self-signed cert in-memory
... caches, leader election, certs and so on ...
I0721 13:04:46.081860  441897 leaderelection.go:258] successfully acquired lease openshift-cloud-controller-manager/cloud-controller-manager
I0721 13:04:46.082182  441897 controllermanager.go:164] using static initialization from config file /tmp/tel-qrzafe31/fs/etc/kubernetes-cloud-config/cloud.conf
I0721 13:04:46.082283  441897 controllermanager.go:256] Version: v0.0.0-master+$Format:%h$
I0721 13:04:47.996900  441897 azure_auth.go:235] Using AzureStackCloud environment
I0721 13:04:48.884379  441897 azure_auth.go:120] azure: using client_id+client_secret to retrieve access token
I0721 13:04:48.884416  441897 azure.go:678] Azure cloudprovider using try backoff: retries=6, exponent=1.500000, duration=6, jitter=1.000000
I0721 13:04:48.884480  441897 azure_diskclient.go:67] Azure DisksClient using API version: 2019-03-01
I0721 13:04:48.884566  441897 azure_zones.go:46] refreshZones: starting to fetch all available zones for the subscription e90a4bc2-2f4e-4d36-874b-9011e4734d74
W0721 13:04:50.602066  441897 azure_zones.go:53] refreshZones: empty zone list, will retry after 30m0s
E0721 13:04:50.602134  441897 azure.go:579] InitializeCloudFromConfig: failed to sync regional zones map for the first time: empty zone list
F0721 13:04:50.602316  441897 controllermanager.go:271] Cloud provider azure could not be initialized: could not init cloud provider azure: empty zone list
goroutine 229 [running]:
k8s.io/klog/v2.stacks(0xc000192001, 0xc0001d8460, 0x9c, 0xd7)
... HUUUUGE PANIC TRACE BELOW
```
